scene/gui/tab_bar.cpp

```diff
@@ -586,7 +685,12 @@ String TabBar::get_tab_language(int p_tab) const {
 void TabBar::set_tab_icon(int p_tab, const Ref<Texture2D> &p_icon) {
 	ERR_FAIL_INDEX(p_tab, tabs.size());
 	tabs.write[p_tab].icon = p_icon;
+
 	_update_cache();
+	_ensure_no_over_offset();
+	if (scroll_to_selected) {
+		ensure_tab_visible(current);
+	}
 	update();
 	update_minimum_size();
```

Comment:
Seeing as this code is repeated many times, maybe it could be moved to a method? Something like `refresh_all()` idk.

Reply:
Those group of functions don't always carry all those functions, as some would be unnecessary steps in some occasions. So there isn't a "one size fits it all" to throw into a single function.

Reply:
Searching this brings 8 exact results, but ok.